src/tools/bin/plot_chargeConservation_overTime.py

```diff
@@ -0,0 +1,209 @@
+
+    # load and add charge density
+    charge = np.zeros_like(Ex)
+    norm = 0.0
+    for field_name in f["/data/{}/fields/".format(timestep)].keys():
+        if field_name[0:8] == "Density_":
+            # load species density
+            species_Density = np.array(f["/data/{}/fields/".format(timestep) + field_name])
+            # choose norm to be the maximal charge density of all species
+            norm = np.max([norm, np.amax(np.abs(species_Density))])
+            # add charge density to total charge density
+            charge += species_Density
+
+    # close hdf5 file
+    f.close()
+
+    # compute divergence of electric field according to Yee scheme
+    div = ((Ex[1:, 1:, 1:] - Ex[1:, 1:, :-1])/CELL_WIDTH +
+           (Ey[1:, 1:, 1:] - Ey[1:, :-1, 1:])/CELL_HEIGHT +
+           (Ez[1:, 1:, 1:] - Ez[:-1, 1:, 1:])/CELL_DEPTH)
+
+    # compute difference between electric field divergence and charge density
+    diff = (div  - charge[1:, 1:, 1:]/EPS0)
+
+    return float(timestep), np.amax(np.abs(diff)), np.mean(np.abs(diff)), np.std(diff), norm
+
+
+# ---- main program ----------
+
+if __name__ == "__main__":
+    # set up argument parser
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        epilog='For further questions please contact Richard Pausch.'
+        )
+
+    parser.add_argument(metavar="simulation directory",
+                        dest="directory",
+                        help='simulation base directory',
+                        action='store')
+
+    args = parser.parse_args()
+    directory = args.directory
+
+    # do the data reading and catch errors
+    try:
+        # test if directory is a directory
+        if not os.path.isdir(directory):
+            raise Exception('Error: {} is not a directory'.format(directory))
+
+        # check if any hdf5 files were found
+        h5_file_list = get_list_of_hdf5_files(directory)
+        if len(h5_file_list) == 0:
+            raise Exception('No hdf5 files found in {}'.format(directory+"simOutput/"))
+
+    except Exception as error_msg:
+        print("{}".format(error_msg))
+        sys.exit(1)
+
+    # collect data from all found hdf5 files
+    collect_results = None
+    print("Read files:")
+    for f in h5_file_list:
+        print(f)
+        t, cc_max, mean_abs, std, norm =  deviation_charge_conservation(f)
+        data_tmp = np.array([[t, cc_max, mean_abs, std, norm]])
+        if type(collect_results) == type(None):
+            collect_results = data_tmp
+        else:
+            collect_results = np.append(collect_results, data_tmp, axis=0)
+
+    # sort data temporally
+    collect_results = np.sort(collect_results, axis=0)
+
+    # alias to data
+    t = collect_results[:, 0] # all timesteps
+    max_diff = collect_results[:, 1] # all max abs diff
+    mean_abs = collect_results[:, 2] # all mean abs
+    std = collect_results[:, 3] # all std
+    norm = collect_results[0, 4] # first (t=0) norm
```

Comment:
exactly `collect_result[:, 4]` is the `np.amax` for each time steps, this `collect_result[:, 4]` is the `np.amax` of the first time step.